### PR TITLE
Add axe analysis to pipeline

### LINKED ISSUE #5402: Add axe to pipeline
Add axe analysis to pipeline

Out of scope, followup tickets:
- [x] #6354
- [x] #6355

### PR DESCRIPTION
**Pull request by wkoot**
Closes #5402

### CODE REVIEW

File: docs/src/changelog.md [hunk 1]
```diff
@@ -48,6 +48,7 @@ See the [deployment instructions](https://quality-time.readthedocs.io/en/latest/
 ### Added
 
 - When using SonarQube security hotspots as source for the security warnings metric, allow for filtering hotspots by hotspot status ('to review', 'acknowledged', 'fixed', 'safe'). The default value of this parameter is to show hotspots 'to review' and 'acknowledged'. Closes [#5956](https://github.com/ICTU/quality-time/issues/5956).
+- Add aXe analysis to Quality-time pipeline. Closes [#5402](https://github.com/ICTU/quality-time/issues/5402).
```
Comment (fniessink) [inline docs/src/changelog.md]: What's with the weird spelling? Axe writes axe as `axe` on their website.

```suggestion
- Add axe analysis to Quality-time pipeline. Closes [#5402](https://github.com/ICTU/quality-time/issues/5402).
```

File: tests/application_tests/src/test_report.py [hunk 2]
```diff
@@ -88,3 +90,28 @@ def test_add_report(self):
         nr_reports = len(self.driver.find_elements(By.CLASS_NAME, "card"))
         self.driver.find_element(By.CLASS_NAME, "button.primary").click()
         self.wait.until(nr_elements((By.CLASS_NAME, "card"), nr_reports + 1))
+
+    def test_report_axe_accessibility(self):
+        """Run aXe accessibility check on a report."""
```
Comment (fniessink) [inline tests/application_tests/src/test_report.py]:
```suggestion
        """Run axe accessibility check on a report."""
```

File: tests/application_tests/src/test_report.py [hunk 3]
```diff
@@ -88,3 +90,28 @@ def test_add_report(self):
...
+        filename = pathlib.Path('../../build/a11y_violations.txt')
+        try:
+            with open(filename, "w", encoding="utf8") as report_file:
+                report_file.write(readable_report)
+        except OSError:
+            print("Could not write axe violations txt")
+
+        # If there are violations, output the readable report data
+        # TODO - enable in https://github.com/ICTU/quality-time/issues/6354
+        # self.assertEqual(0, len(violation_results), readable_report)
```
Comment (fniessink) [inline tests/application_tests/src/test_report.py]: Can we maybe add an assert that the number of violations is <= the current number of violations?

**Review by fniessink — CHANGES_REQUESTED** [review]

File: docs/src/changelog.md (hunk 1 above)
Comment (wkoot) [reply docs/src/changelog.md]: Huh, peculiar, I don't know how I picked this up

**Review by fniessink — APPROVED** [review]

**Pull request merged**